### PR TITLE
fix warnings

### PR DESCRIPTION
**Pull request by petabytestorage**
```	--enable=unused \
   	--enable=varcheck \
	--enable=vetshadow \
	--enable=vet \
	./...
WARNING: staticcheck, gosimple and unused are all set, using megacheck instead
pubsub/query/query_test.go:74::error: github.com/tendermint/tmlibs/pubsub/query.Condition composite literal uses unkeyed fields (vet)
pubsub/query/query_test.go:75::error: github.com/tendermint/tmlibs/pubsub/query.Condition composite literal uses unkeyed fields (vet)
pubsub/query/query_test.go:75::error: github.com/tendermint/tmlibs/pubsub/query.Condition composite literal uses unkeyed fields (vet)
pubsub/query/query_test.go:76::error: github.com/tendermint/tmlibs/pubsub/query.Condition composite literal uses unkeyed fields (vet)
Makefile:31: recipe for target 'metalinter_test' failed
make[1]: *** [metalinter_test] Error 1
make[1]: Leaving directory '/home/pbs/go/src/github.com/tendermint/tmlibs'
Makefile:13: recipe for target 'test' failed
make: *** [test] Error 2
```

### CODE REVIEW

File: pubsub/query/parser_test.go [hunk 1]
```diff
@@ -83,9 +83,9 @@ func TestParser(t *testing.T) {
 	for _, c := range cases {
 		_, err := query.New(c.query)
 		if c.valid {
-			assert.NoError(t, err, "Query was '%s'", c.query)
+			assert.NoErrorf(t, err, "Query was '%s'", c.query)
```
Comment (ebuchman) [inline pubsub/query/parser_test.go]: not a real method. same below

Comment (petabytestorage) [reply pubsub/query/parser_test.go]: hard for me to understand why `make` and then tests seem to pass but i believe you @ebuchman

Comment (melekes) [reply pubsub/query/parser_test.go]: https://github.com/stretchr/testify/blob/master/assert/assertion_format.go#L236

Comment (melekes) [reply pubsub/query/parser_test.go]: what triggered the change? `assert.NoErrorf(t, err, "Query was '%s'", c.query)` this should also compile, no?

Comment (petabytestorage) [reply pubsub/query/parser_test.go]: this `go vet` error @melekes   ```	./...
WARNING: staticcheck, gosimple and unused are all set, using megacheck instead
pubsub/query/parser_test.go:88::error: possible formatting directive in Error call (vet)
``` goes away when adding `f` at the end similar to @odeke-em solutions elsewhere i believe

**Comment by melekes** [comment]
Tests are failing with:

```
WARNING: staticcheck, gosimple and unused are all set, using megacheck instead
pubsub/query/parser_test.go:86:4:warning: unused variable or constant NoErrorf not declared by package assert (varcheck)
pubsub/query/parser_test.go:88:4:warning: unused variable or constant Errorf not declared by package assert (varcheck)
pubsub/query/parser_test.go:86:4:warning: NoErrorf not declared by package assert (megacheck)
```

**Comment by petabytestorage** [comment]
i'm running `go vet` with `go version 1.9.2`   anybody else on 1.9.x?

**Comment by petabytestorage** [comment]
i found that after running `glide update` i get the same problems that this has. so i must have been on old deps. should close this PR and will reopen different one to fix later if necessary. now it looks like strethr testify changed.

**Comment by petabytestorage** [comment]
found a fix ty for help

File: pubsub/query/parser_test.go [hunk 2]
```diff
@@ -83,9 +83,9 @@ func TestParser(t *testing.T) {
 	for _, c := range cases {
 		_, err := query.New(c.query)
 		if c.valid {
-			assert.NoError(t, err, "Query was '%s'", c.query)
+			assert.NoError(t, err, "Query was '"+c.query+"'")
```
Comment (melekes) [inline pubsub/query/parser_test.go]: can't we use `assert.NoErrorf`?

Comment (petabytestorage) [reply pubsub/query/parser_test.go]: https://github.com/tendermint/tmlibs/issues/92

**Pull request merged**